sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects.scala

```diff
@@ -680,3 +680,54 @@ case class AssertNotNull(child: Expression, walkedTypePath: Seq[String])
      """
   }
 }
+
+case class GetExternalRowField(
```

Comment:
You can take a look at `GetStructFiled`, which gets field from internal row, and is similar to this one.